azure/services/securitygroups/securitygroups.go

```diff
@@ -77,10 +77,10 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			// Check if the expected rules are present
 			update := false
 			securityRules = *existingNSG.SecurityRules
-			for _, rule := range nsgSpec.IngressRules {
-				if !ruleExists(securityRules, converters.IngresstoSecurityRule(rule)) {
+			for _, rule := range nsgSpec.SecurityRules {
+				if !ruleExists(securityRules, converters.SecurityRuleToSDK(rule)) {
```

Comment:
Maybe worth storing the result of `converters.SecurityRuleToSDK(rule)` to a local variable to avoid calling it twice?